fix(table): fix accessibility for mobile

christiemolloy opened the pull request:
closes #2540

patternfly-build commented:
PatternFly-Next preview: https://patternfly-next-pr-2582.surge.sh

mattnolting commented on src/patternfly/components/Table/table-td.hbs:
```diff
@@ -6,6 +6,7 @@
     {{else if table-td--action}} class="pf-c-table__action"
     {{else if table-td--modifier}} class="{{table-td--modifier}}"
     {{/if}}
+    role="cell"
```
Within `.pf-c-table__expandable-row.pf-m-expanded`, if empty cells are present, VO announces them where it should ignore them. Omitting `role="cell"` attr on them hides. 

```
{{#unless table-td--IsEmpty}}
  role="cell"
{{/unless}}
```

christiemolloy replied:
Oh yes good point!

christiemolloy replied:
updated

mattnolting replied:
![Screen Shot 2020-01-21 at 1 38 21 PM](https://user-images.githubusercontent.com/5385435/72832656-50933c00-3c53-11ea-9c92-ed78d07fd82e.png)

https://github.com/patternfly/patternfly-next/blob/master/src/patternfly/components/Table/table-td.hbs#L9
```
{{#unless table-td--IsEmpty}}
  role="cell"
{{/unless}}
```

then in [table.md](https://github.com/patternfly/patternfly-next/blob/master/src/patternfly/components/Table/examples/Table.md) add `table-cell-IsEmpty=“true”`

mattnolting replied:
Then in [table.md](https://github.com/patternfly/patternfly-next/blob/master/src/patternfly/components/Table/examples/Table.md), anywhere you see `{{#> table-td}}{{/table-td}}` (lines 257, 408, 409, 441, 442, 602, 603, 834, 835, 1150, 1151, 1180, 1181, 1187) add `table-td--IsEmpty="true"`. 
lines: 996 & 997 (should be `table-td table-td--IsEmpty="true"` instead of `table-th`)

christiemolloy replied:
This was really helpful feedback thanks @mattnolting

mcoker commented:
Looks great, though I see these discrepancies in some of the examples. Is that on purpose?

* compact & compact w/o borders - wrong role on empty cells
* hidden/vislble missing scope and role
* headers that wrap missing scope

christiemolloy commented:
@mcoker thanks for finding those inconsistencies. I attempted to address them by doing this:
- Matt Noltings comment addressed this problem
- Looks like this table was missing table--grid="true", added that in and added `scope="col"`
- There was no `scope="col"` set on hidden/visible, added that in

Could you confirm that these have been fixed?

redallen commented:
:tada: This PR is included in version 2.51.1 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@patternfly/patternfly)
- [GitHub release](https://github.com/patternfly/patternfly-next/releases/tag/v2.51.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
